src/objects/datetime.rs

```diff
@@ -0,0 +1,305 @@
+use err::PyResult;
+use ffi::PyDateTimeAPI;
+use ffi::{PyDateTime_Check, PyDateTime_DateTimeType};
+use ffi::{
+    PyDateTime_DATE_GET_HOUR, PyDateTime_DATE_GET_MICROSECOND, PyDateTime_DATE_GET_MINUTE,
+    PyDateTime_DATE_GET_SECOND,
+};
+use ffi::{
+    PyDateTime_DELTA_GET_DAYS, PyDateTime_DELTA_GET_MICROSECONDS, PyDateTime_DELTA_GET_SECONDS,
+};
+use ffi::{
+    PyDateTime_DateTime_GET_DAY, PyDateTime_DateTime_GET_MONTH, PyDateTime_DateTime_GET_YEAR,
+};
+use ffi::{PyDateTime_DateType, PyDate_Check};
+use ffi::{PyDateTime_Date_GET_DAY, PyDateTime_Date_GET_MONTH, PyDateTime_Date_GET_YEAR};
+use ffi::{PyDateTime_DeltaType, PyDelta_Check};
+use ffi::{
+    PyDateTime_TIME_GET_HOUR, PyDateTime_TIME_GET_MICROSECOND, PyDateTime_TIME_GET_MINUTE,
+    PyDateTime_TIME_GET_SECOND,
+};
+use ffi::{PyDateTime_TZInfoType, PyTZInfo_Check};
+use ffi::{PyDateTime_TimeType, PyTime_Check};
+use object::PyObject;
+use std::os::raw::c_int;
+
+#[cfg(Py_3_6)]
+use ffi::{PyDateTime_DATE_GET_FOLD, PyDateTime_TIME_GET_FOLD};
+
+use instance::Py;
+use python::{Python, ToPyPointer};
+
+// Traits
+pub trait PyDateComponentAccess {
```

Comment:
That's a bit of a bikeshed, but imo we could remove the `Component` from the name. A small doc comment like `Gives access to year, month and day` would also nice for rustdoc.

Reply:
Yeah, it was long enough that I was considering moving it into a namespace, like:

```rust
pub mod  Components {
pub trait PyDateAccess { ... }
}
```

Either way I'll remove `Component` from the name.